test/k8sT/assertionHelpers.go

```diff
@@ -163,6 +163,48 @@ func RedeployCiliumWithMerge(vm *helpers.Kubectl,
 	RedeployCilium(vm, ciliumFilename, newOpts)
 }
 
+// optionChangeRequiresPodRedeploy returns true if the difference between the
+// specified options requires redeployment of all pods to ensure that the
+// datapath is operating consistently.
+func optionChangeRequiresPodRedeploy(prev, next map[string]string) bool {
```

Comment:
I think this is a dead code, i.e., nobody invokes the function.

Reply:
That is correct. Removing the commit.

Reply:
Actually, maybe we should introduce the invoker? Are we hitting the issues this is fixing on v1.9? Moving this discussion to #18031.

Reply:
Added #16767 and #16835 to this backport as #18031 depends on them. See discussion at https://github.com/cilium/cilium/pull/18031#issuecomment-991084995.